dpnp/dpnp_algo/dpnp_algo_indexing.pyx

```diff
@@ -184,12 +185,18 @@ cpdef tuple dpnp_nonzero(dparray in_array1):
 
 
 cpdef dpnp_place(dparray arr, dparray mask, vals):
-    cpdef int counter = 0
     cpdef int vals_len = len(vals)
-    for i in range(arr.size):
-        if mask[i]:
-            arr[i] = vals[counter % vals_len]
-            counter += 1
+    vals_arr = dparray(vals_len, dtype=arr.dtype)
+    for i in range(vals_len):
```

Comment:
I think it would be better to move this code into the kernel. I see no algorithm problems here